src/test/groovy/com/palantir/gradle/gitversion/GitVersionPluginTests.groovy

```diff
@@ -151,6 +141,45 @@ class GitVersionPluginTests extends Specification {
         buildResult.standardOutput.contains(':printVersion\n1.0.0.dirty\n')
     }
 
+    def 'git describe long when annotated tag is present and long is true' () {
+        given:
+        buildFile << '''
+            plugins {
+                id 'com.palantir.git-version'
+            }
+            gitVersionDescriber().setLong(true)
```

Comment:
I think the mutability here is pretty undesirable, as some invocations may make it much harder (especially in multi project builds) to keep calls to `gitVersion()` idempotent.